modules/carto/src/api/maps-api-client.js

```diff
@@ -84,83 +75,50 @@ function dealWithWindshaftError({response, layergroup, credentials}) {
           credentials.apiKey
         }') doesn't provide access to the requested data`
       );
+
     default:
-      throw new Error(`${JSON.stringify(layergroup.errors)}`);
+      const e = getMapsVersion() === 'v1' ? JSON.stringify(json.errors) : json.error;
+      throw new Error(e);
   }
 }
 
 /**
- * Create a GET or POST request, with all required parameters
+ * Build a URL all required parameters
  */
-function createMapsApiRequest({config, credentials}) {
+function buildURL({mapConfig, credentials}) {
+  const cfg = JSON.stringify(mapConfig);
   const encodedApiKey = encodeParameter('api_key', credentials.apiKey);
   const encodedClient = encodeParameter('client', `deck-gl-carto`);
```

Comment:
"deck-gl-carto", could be also a parameter in the configuration, that could be injected from outside or have this default value